src/core/lib/surface/call.cc

```diff
@@ -321,16 +321,24 @@ grpc_error* grpc_call_create(const grpc_call_create_args* args,
 
   GRPC_CHANNEL_INTERNAL_REF(args->channel, "call");
 
+  grpc_core::Arena* arena;
+  grpc_call* call;
   grpc_error* error = GRPC_ERROR_NONE;
   grpc_channel_stack* channel_stack =
       grpc_channel_get_channel_stack(args->channel);
-  grpc_call* call;
   size_t initial_size = grpc_channel_get_call_size_estimate(args->channel);
   GRPC_STATS_INC_CALL_INITIAL_SIZE(initial_size);
-  grpc_core::Arena* arena = grpc_core::Arena::Create(initial_size);
-  call = new (arena->Alloc(GPR_ROUND_UP_TO_ALIGNMENT_SIZE(sizeof(grpc_call)) +
-                           channel_stack->call_stack_size))
-      grpc_call(arena, *args);
+  size_t call_and_stack_size =
+      GPR_ROUND_UP_TO_ALIGNMENT_SIZE(sizeof(grpc_call)) +
+      channel_stack->call_stack_size;
+  size_t call_alloc_size =
+      call_and_stack_size + (args->parent ? sizeof(child_call) : 0);
+
+  std::pair<grpc_core::Arena*, void*> arena_with_call =
+      grpc_core::Arena::CreateWithAlloc(initial_size, call_alloc_size);
+  arena = arena_with_call.first;
+  void* call_storage = arena_with_call.second;
```

Comment:
please don't alias with void*. this constraints the compiler optimize. instead use it directly:
```
call = new (arena_with_call.second) grpc_call(arena, *args);
```

Reply:
But arena_with_call.second is also a void *. Should I change that to a char*? (The pair<Arena*, void*> I mean).

Reply:
Now what I mean is that when you alias the pointer as a local variable it adds a constraint on that variable because it can alias everything. so it's recommended not to create a local variable unless it's used more than once. No need to change the type.